Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -254,16 +246,8 @@ private bool TryDeserializeMethod(MethodInfo method, IIonReader reader, IonType
             // We deserialize whether or not we ultimately use the result because values
             // for some Ion types need to be consumed in order to advance the reader.
             deserialized = ionSerializer.Deserialize(reader, property.PropertyType, ionType);
-            
-            if (IsReadOnlyProperty(property))
```

Comment:
Why is this removed? Won't the deserialize logic attempt to set the value to property?

Reply:
Thanks to the code coverage, I discovered that this condition is never satisfied (because readonly properties are previously filtered out by `GetValidProperties`).